### PR TITLE
feat: add GetRolesForUserForAllDomains

### PR DESCRIPTION
**Pull request by 00LT00**
Fix: https://github.com/casbin/casbin/issues/588

### CODE REVIEW

**Comment by 00LT00** [comment]
plz review @PhilippSeitz @hsluoyz @nodece

File: rbac/default-role-manager/role_manager.go [hunk 1]
```diff
@@ -193,6 +193,18 @@ func (rm *RoleManager) GetRoles(name string, domain ...string) ([]string, error)
 	return roles, nil
 }
 
+func (rm *RoleManager) GetAllDomainsForUser(name string) []string {
+	var res []string
+	rm.allDomains.Range(func(key, value interface{}) bool {
+		role := value.(*Roles).createRole(name, rm.matchingFunc)
```
Comment (PhilippSeitz) [inline rbac/default-role-manager/role_manager.go:199]: is the `matchingFunc` supposed to work already, like discussed here #588?

```
p, admin, *, read, allow
p, admin, *, write, allow
p, member, *, read, allow
p, trudy.custom, trudy_org, write, allow
p, trudy.custom, trudy_org, read, deny

g, alice, admin, alice_org
g, bob, member, alice_org
g, trudy, member, trudy_org
g, trudy, trudy.custom, trudy_org
```

```go
e, _ := casbin.NewEnforcer(config)
e.GetRoleManager().(*defaultrolemanager.RoleManager).AddMatchingFunc("keyMatch", util.KeyMatch)
result, err := e.GetImplicitPermissionsForUserForAllDomains(subject)
expected := [][]string{
	{"member", "trudy_org", "read", "allow"},
	{"trudy.custom", "trudy_org", "write", "allow"},
	{"trudy.custom", "trudy_org", "read", "deny"},
}
Expect(err).ShouldNot(HaveOccurred())
Expect(result).Should(ConsistOf(expected))
```

will result in:
```
  Expected
      <[][]string | len:2, cap:2>: [
          ["trudy.custom", "trudy_org", "write", "allow"],
          ["trudy.custom", "trudy_org", "read", "deny"],
      ]
  to consist of
      <[]interface {} | len:1, cap:1>: [
          [
              ["member", "trudy_org", "read", "allow"],
              ["trudy.custom", "trudy_org", "write", "allow"],
              ["trudy.custom", "trudy_org", "read", "deny"],
          ],
      ]
  the missing elements were
      <[]interface {} | len:1, cap:1>: [
          ["member", "trudy_org", "read", "allow"],
      ]
```

**Review by PhilippSeitz — CHANGES_REQUESTED** [review]

**Comment by hsluoyz** [comment]
@00LT00 any update?

**Comment by hsluoyz** [comment]
@00LT00 any update?

**Pull request closed without merging**